Automattic/jetpack
# Jetpack Sync: Enforce args size limits per action

Adds args size checks before attempting to add a certain action to the Sync queue.

If the arguments' size turns out to be larger than the max defined, the corresponding action will not be added to the Sync queue.

## Proposed changes:

* `Sync\Defaults`: Introduce a new static property `$default_max_args_size_per_action` that holds the max serialized data size per Sync action.
* `Sync\Listener`: Introduce a new protected method, `check_data_size_by_action` for checking the args size before attempting to add a certain action to the Sync queue.
* `WP_Test_Jetpack_Sync_Listener`: Add 3 new unit tests

### Other information:


## Jetpack product discussion


[p9F6qB-cii-p2](p9dueE-7rF-p2)

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






- Set `Defaults::$default_max_args_size_per_action['updated_option']['default']` = 1; 
- Update the `blogname`
- Confirm the action will not be synced
- Revert the `Defaults::$default_max_args_size_per_action['updated_option']['default']` change
- Update the `blogname` once more
- Confirm the action is synced

Make sure to checkout the added unit tests as well.
